timesketch/api/v1/resources/sigma.py

```diff
@@ -89,4 +95,72 @@ def get(self, rule_uuid):
             abort(
                 HTTP_STATUS_CODE_NOT_FOUND, 'No sigma rule found with this ID.')
 
-        return return_rule
+        meta = {'current_user': current_user.username,
+                'rules_count': len(sigma_rules)}
+        return jsonify({'objects': return_rule, 'meta': meta})
+
+class SigmaByTextResource(resources.ResourceMixin, Resource):
+    """Resource to get a Sigma rule by text."""
+
+    @login_required
+    def post(self):
+        """Handles POST request to the resource.
+
+        Returns:
+            JSON sigma rule
+        """
+
+        form = request.json
+        if not form:
+            form = request.data
+
+        action = form.get('action', '')
+        
+        if action != 'post':
+            return abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                'Action needs to be "post"')
+        if action == 'post':
```

Comment:
this `if` is not required... since you already have a check for `action != 'post'` which means that everything that follows has `action == 'post'`

but why are you checking the `action` this is a `def post(` function, and will not be called unless this is a post request. and why does the post request have to have a `action == 'post'` if it has no meaning?

Reply:
ack, well I would still keep the action in here, as who knows if we have more later on.